src/inmanta/deploy/scheduler.py

```diff
@@ -372,6 +372,139 @@ async def get_facts(self, resource: dict[str, object]) -> None:
             )
         )
 
+    async def get_resource_state(self) -> SchedulerStatusReport:
+        """
+        Check that the state of the resources in the DB corresponds
+        to the internal state of the scheduler and return a SchedulerStatusReport
+        object containing the internal state and any discrepancies between this
+        state and that of the DB.
+        """
+
+        resources_in_db: Mapping[ResourceIdStr, ResourceDetails]
+
+        def _build_discrepancy_map(
```

Comment:
This method may take a long time. I wonder if we should yield to the ioloop ('await sleep(0)`) every now and them to keep the session alive ? (e.g. every 1000 loop iterations) or run this on a thread